packages/block-library/src/table/block.json

```diff
@@ -29,6 +29,11 @@
 							"type": "string",
 							"default": "td",
 							"source": "tag"
+						},
+						"align": {
+							"type": "string",
+							"source": "attribute",
+							"attribute": "data-align"
```

Comment:
Interesting. since it applied as a class, makes me wonder if at some point, we'd want "classNames" sources. That said, it can be complex and I know it's a deliberate decision to restrict the number of sources available. css @aduth

Reply:
Yeah, I think we'd considered once before a source serialized as the presence of a class name. Personally I don't have much of an issue with the `data-` attributes; they can be styled just as well, are a bit easier to parse, and fit well as serving the purpose of a data attribute.

>Custom data attributes are intended to store custom data, state, annotations, and similar, private to the page or application, for which there are no more appropriate attributes or elements.

https://html.spec.whatwg.org/multipage/dom.html#embedding-custom-non-visible-data-with-the-data-*-attributes

That being said, I'm not really sure why we bother to serialize this in the HTML at all, if we're not using it for anything like styling? Why not just embed it within the comment delimiters (i.e. omit `source`)?

Reply:
@aduth As this attribute is in a `query`, my understanding is that `source` can't be omitted. A `classNames` source could be useful in this use case.

An argument against `data-` attributes is that it's double storing data, but then it's not too much of a problem with the reactive nature of blocks, it's not like there's extra business logic to keep them in sync.

Reply:
Why use the class at all? Can we not just use the data attribute?

Reply:
There are some global styles for text alignment:
https://github.com/WordPress/gutenberg/blob/4ac7dc9d2d93a1b3ab81b44a6fbd89952a44915e/packages/block-library/src/style.scss#L150-L161

Seems better to be able to use those and not have extra styles in the table block.